File: app/src/main/java/xyz/sleekstats/completist/view/PersonSummaryDialog.java
package xyz.sleekstats.completist.view;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.TextView;

import xyz.sleekstats.completist.R;

public class PersonSummaryDialog extends DialogFragment {

    private static final String KEY_SUMMARY = "key_summary";

    public PersonSummaryDialog() {
    }

    public static PersonSummaryDialog newInstance(String summary) {

        Bundle args = new Bundle();
        PersonSummaryDialog fragment = new PersonSummaryDialog();
        args.putString(KEY_SUMMARY, summary);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        Bundle bundle = getArguments();
        if(bundle == null) {
            return super.onCreateDialog(savedInstanceState);
        }
        String summary = bundle.getString(KEY_SUMMARY);
        View rootView = LayoutInflater.from(getActivity()).inflate(R.layout.dialog_person_summary, null);
        TextView textView = rootView.findViewById(R.id.dialog_person_summary);
        textView.setText(summary);
        textView.setMovementMethod(new ScrollingMovementMethod());

        AlertDialog alertDialog = new AlertDialog.Builder(getActivity(), AlertDialog.THEME_HOLO_DARK)
                .setView(rootView)
                .setTitle("Summary")
                .setNegativeButton(R.string.back, (dialog, id) -> {
                    if (dialog != null) {
                        dialog.dismiss();
                    }
                })
                .create();
        alertDialog.setCanceledOnTouchOutside(false);
        alertDialog.show();
        return alertDialog;
    }
}

File: app/src/main/java/xyz/sleekstats/completist/view/PersonSummaryDialog.kt
package xyz.sleekstats.completist.view

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.view.LayoutInflater
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import xyz.sleekstats.completist.R

class PersonSummaryDialog : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = arguments ?: return super.onCreateDialog(savedInstanceState)
        val summary = bundle.getString(KEY_SUMMARY)
        @SuppressLint("InflateParams") val rootView = LayoutInflater.from(activity).inflate(R.layout.dialog_person_summary, null)
        val textView = rootView.findViewById<TextView>(R.id.dialog_person_summary)
        textView.text = summary
        textView.movementMethod = ScrollingMovementMethod()
        val alertDialog = AlertDialog.Builder(activity, AlertDialog.THEME_HOLO_DARK)
                .setView(rootView)
                .setTitle("Summary")
                .setNegativeButton(R.string.back) { dialog: DialogInterface?, _: Int ->
                    dialog?.dismiss()
                }
                .create()
        alertDialog.setCanceledOnTouchOutside(false)
        alertDialog.show()
        return alertDialog
    }

    companion object {
        private const val KEY_SUMMARY = "key_summary"
        fun newInstance(summary: String?): PersonSummaryDialog {
            val args = Bundle()
            val fragment = PersonSummaryDialog()
            args.putString(KEY_SUMMARY, summary)
            fragment.arguments = args
            return fragment
        }
    }
}